ush/detect_machine.sh

```diff
@@ -1,9 +1,21 @@
 #!/bin/bash
 
+# The authoritative copy of this script lives in the ufs-weather-model at:
+# https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/detect_machine.sh
+# If any local modifications are made or new platform support added,
+# please consider opening an issue and a PR to the ufs-weather-model
+# so that this copy remains in sync with its authoritative source
+#
+# Thank you for your contribution
+
+# If the MACHINE_ID variable is set, skip this script.
+[[ -n ${MACHINE_ID:-} ]] && return
```

Comment:
This should probably properly be `-v MACHINE_ID`, but since the goal is to match UFS, it's fine.